File: backend-java/auth-service/src/main/java/com/cdm/auth/entity/PermissionEntity.java
package com.cdm.auth.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "permissions")
@Getter @Setter @NoArgsConstructor
public class PermissionEntity extends BaseEntity {

    @Column(nullable = false, length = 100)
    private String name;

    @Column(nullable = false, unique = true, length = 100)
    private String code;

    @Column(name = "resource_id")
    private Long resourceId;

    @Column(name = "action_id")
    private Long actionId;
}

File: backend-nestjs/apps/auth/src/rbac/permission.entity.ts
import { Entity, Column, PrimaryColumn } from 'typeorm';

@Entity('permissions')
export class PermissionEntity {
  @PrimaryColumn({ type: 'bigint' })
  id: string;

  @Column({ length: 100 })
  name: string;

  @Column({ unique: true, length: 100 })
  code: string;

  @Column({ name: 'resource_id', type: 'bigint' })
  resourceId: string;

  @Column({ name: 'action_id', type: 'bigint' })
  actionId: string;
}
